src/GSoft.Extensions.MediatR.Analyzers.Tests/CompilationHelper.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace GSoft.Extensions.MediatR.Analyzers.Tests;
+
+// This class can be copy-pasted and reused as-is in other Roslyn Analyzers test projects
+internal sealed class CompilationHelper
```

Comment:
Very useful class that simulates a sln + csproj compilation in-memory and returns the diagnostics (warnings and errors).

It's actually the same APIs that are used when we compile a project or solution in the real world.

Reply:
I thought about that... let's see how much we actually need to reuse this before considering creating another package.

I don't expect us to create that much Roslyn analyzers in the future TBH. Not everybody have the knowledge, and it's easy to write slow analyzers.

Reply:
Also we might end up with too many packages to manage 😬